Detectors/MUON/MCH/Workflow/src/DataDecoder.cxx

```diff
@@ -0,0 +1,602 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    DatDecoder.cxx
+/// \author  Andrea Ferrero
+///
+/// \brief Implementation of a data processor to run the raw decoding
+///
+
+#include <random>
+#include <iostream>
+#include <fstream>
+#include <stdexcept>
+#include "Framework/CallbackService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/ControlService.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Lifetime.h"
+#include "Framework/Output.h"
+#include "Framework/Task.h"
+#include "Framework/WorkflowSpec.h"
+
+#include "DPLUtils/DPLRawParser.h"
+#include "MCHBase/Digit.h"
+#include "Headers/RAWDataHeader.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "MCHRawCommon/DataFormats.h"
+#include "MCHRawDecoder/PageDecoder.h"
+#include "MCHRawElecMap/Mapper.h"
+#include "MCHMappingInterface/Segmentation.h"
+#include "MCHWorkflow/DataDecoder.h"
+
+static bool mPrint = false;
+
+namespace o2
+{
+namespace mch
+{
+namespace raw
+{
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::mch::mapping;
+using RDHv4 = o2::header::RAWDataHeaderV4;
+
+static std::array<int, 64> refManu2ds_st345_v5 = {
```

Comment:
please add some comments explaining the difference between those two arrays

Reply:
Done.